core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -317,6 +320,37 @@ object Uri {
       val merge = segments.initLast.map(_._1).getOrElse(segments)
       Path(merge.concat(path.segments), absolute = absolute, endsWithSlash = path.endsWithSlash)
     }
+
+    def startsWith(path: Path): Boolean = {
+      val i = segments.iterator
+      val j = path.segments.iterator
+      while (j.hasNext && i.hasNext) if (i.next() != j.next())
+        return false
+
+      !j.hasNext
+    }
+
+    def startsWith_(path: String): Boolean = startsWith(Path.fromString(path))
+
+    def indexOf(path: Path): Option[Int] = {
+      val iter = segments.iterator.zipWithIndex
+      val other = path.segments.iterator
+      var idx: Option[Int] = None
+      while (iter.hasNext && other.hasNext) {
```

Comment:
I'm getting sleepy, but wouldn't this return the last indexOf instead of the first indexOf?

Reply:
possible. I need to add some tests here

Reply:
Replaced Chain with List, so we got this for free